apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -737,13 +748,33 @@ export class WorkshopForm extends React.Component {
     return value;
   };
 
+  currentVirtualStatus = () => {
```

Comment:
I don't love this, if anyone has suggestions.

Reply:
This seems correct.  I know it's a bit complicated, but hopefully we don't have to preserve this logic forever.